Makefile

```diff
@@ -44,6 +46,11 @@ BUILDFLAGS :=  -ldflags \
 # Some tests expect default values for version.*, so just use the config package values there.
 TEST_BUILDFLAGS :=  -ldflags "$(BUILD_TIME_CONFIG_FLAGS)"
 
+ifneq ($(GO_VERSION),$(CATALOG_GO_VERSION))
```

Comment:
Thinking abt this a bit more, I think this can lead to maintenance issues, and builds failing unnecessarily. So let's remove this check, but keep CATALOG_GO_VERSION (may be rename it to DUMMY_GO_VERSION to avoid any confusion) with a comment
```
This version is just used to trigger a new build in case we update the version of catalog, and dont have new PRs which use the updated version in the catalog. This does not reflect the go binary version which was used to build the jx binary 
```

Reply:
Done